File: app/src/main/java/com/afoxxvi/alopex/listener/BootReceiver.java
package com.afoxxvi.alopex.listener;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import com.afoxxvi.alopex.MainActivity;

public class BootReceiver extends BroadcastReceiver {

    private static final String ACTION_BOOT_COMPLETED = "android.intent.action.BOOT_COMPLETED";

    @Override
    public void onReceive(Context context, Intent intent) {
        if (ACTION_BOOT_COMPLETED.equals(intent.getAction())) {
            Intent intent1 = new Intent(context, MainActivity.class);
            intent1.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
            context.startActivity(intent1);
        }
    }
}

File: app/src/main/java/com/afoxxvi/alopex/listener/BootReceiver.kt
package com.afoxxvi.alopex.listener

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import com.afoxxvi.alopex.MainActivity

class BootReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        if (ACTION_BOOT_COMPLETED == intent.action) {
            val intent1 = Intent(context, MainActivity::class.java)
            intent1.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
            context.startActivity(intent1)
        }
    }

    companion object {
        private const val ACTION_BOOT_COMPLETED = "android.intent.action.BOOT_COMPLETED"
    }
}